modules/github_integration/lib/open_project/github_integration/notification_handler/issue_comment.rb

```diff
@@ -0,0 +1,98 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module OpenProject::GithubIntegration
+  module NotificationHandler
+    ##
+    # Handles GitHub issue comment notifications.
+    class IssueComment
+      include OpenProject::GithubIntegration::NotificationHandler::Helper
+
+      COMMENT_ACTIONS = %w[
+        created
+        edited
+      ].freeze
+
+      def process(payload)
+        return unless associated_with_pr?(payload)
+
+        github_system_user = User.find_by(id: payload.fetch('open_project_user_id'))
```

Comment:
How does the OpenProject specific property find its way into the GitHub webhook payload. Is it something that needs to be configured? If we forfeit supporting GitHub users for now, do we still need it?

Reply:
we need it to make sure these requests are legitimate (and not someone throwing curl requests at our openproject instance). Because each webhook request has the github_user token with it, we

1. find the system users for which to create journal entries and verify access to work_packages (so the integration only handles work packages which it has the permission for)
2. know that the request comes from GitHub and not some random dude from the internet